test: refactors test-tls-client-verify

homosaur opened the pull request:
<!--
Thank you for your pull request. Please review below requirements.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j8 test` (UNIX), or `vcbuild test nosign` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
test

##### Description of change
<!-- Provide a description of the change below this comment. -->
Refactor of test-tls-client-verify for NINA Code and Learn

Trott commented on test/parallel/test-tls-client-verify.js:
```diff
@@ -71,45 +71,52 @@ function testServers(index, servers, clientOptions, cb) {
-  if (!tcase) return;
+  const tcase = testCases[testIndex];
+  if (!tcase) {
+    common.mustCall(function() {
```
This doesn't look right. If you put `common.mustCall()` inside a conditional, then the function won't be required to be called unless the conditional is true. But this is an immediately-invoked function, so the `common.mustCall()` seemingly serves no purpose here. Maybe what is meant is for `thisMustRunOnce = common.mustCall(function() {...});` to be outside the conditional so that it always runs, and then inside the conditional: `thisMustRunOnce()`.

homosaur commented:
@Trott, I reverted the final refactor of process.exit to the mustCall pattern (for now) and changed the commit message accordingly. Thanks for the help on it.

Trott commented:
LGTM if CI is ✅

Trott commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/5295/

addaleax commented:
Landed in fd6999ef6cb4a2f865f0d7b94cad0e00bdb24820, thanks for the contribution!
